### PR TITLE
Use grub2 in ISOs

### LINKED ISSUE #521: Use GRUB instead of syslinux in the iso live image
It's much easier to consistently use the same bootloader in all places, and also GRUB is much more flexible in terms of #520 

**Is your feature request related to a problem? Please describe.**
Currently we use syslinux to build our ISOs, using GRUB would help us in the long-term to have a consistent mechanism across live and the installed system.

**Describe the solution you'd like**
Using GRUB in place of syslinux, with the documentation updated as needed

### PR DESCRIPTION
**Pull request by davidcassany**
This commit refactors grub2 usage in cOS. It adds a collection of
packages related to grub2 and configures ISOs generation to make
use of grub2 packages.

Fixes #521

Signed-off-by: David Cassany <dcassany@suse.com>

### CODE REVIEW

**Review by mudler — CHANGES_REQUESTED** [review]
Awesome! It looks good, but there is no bump to cOS, so tests  are hitting the hold grub configurations!

**Comment by davidcassany** [comment]
> Awesome! It looks good, but there is no bump to cOS, so tests are hitting the hold grub configurations!

Ouch! you are right... 😓

**Review by mudler — APPROVED** [review]
Looks good here! let's just wait for CI to pass

File: packages/grub2/build.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+requires:
+- name: "base"
+  category: "distro"
+  version: ">=0"
+
+steps:
+{{if eq .Values.name "grub2-config"}}
```
Comment (Itxaka) [inline packages/grub2/build.yaml:7]: very nice!

**Review by Itxaka — APPROVED** [review]

File: make/Makefile.test [hunk 2]
```diff
@@ -124,7 +124,7 @@ else
 	# config vm options, i.e. efi boot, nat, serial to file, etc...
 	VBoxManage modifyvm "test" --firmware efi --uart1 0x3f8 4 --uartmode1 file $(ROOT_DIR)/serial_port1.log --nic1 nat --boot1 disk --natpf1 "guestssh,tcp,,2222,,22"
 	# add sata controller
-	VBoxManage storagectl "test" --name "sata controller" --add sata
+	VBoxManage storagectl "test" --name "sata controller" --add sata --portcount 2
```
Comment (mudler) [inline make/Makefile.test:127]: One of the quirks of vbox I suppose :sweat_smile:

**Pull request merged**